dig_test.go

```diff
@@ -438,6 +438,16 @@ func TestEndToEndSuccess(t *testing.T) {
 		require.NoError(t, c.Provide(func(A) B { return B{} }), "provide failed")
 		require.NoError(t, c.Provide(func(A, B) C { return C{} }), "provide failed")
 	})
+
+	t.Run("primitives", func(t *testing.T) {
+		c := New()
+		require.NoError(t, c.Provide(func() string { return "piper" }), "string provide failed")
+		require.NoError(t, c.Provide(func() int { return 42 }), "int provide failed")
```

Comment:
Maybe it is out of scope, but can you add a case for time.Duration, which is an alias to int64?